library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -673,6 +679,9 @@ private void updateDatabaseFromHeaders(State state) {
             values.put(DownloadContract.Downloads.COLUMN_MIME_TYPE, state.mimeType);
         }
 
+        originalDownloadInfo.setETag(state.headerETag);
```

Comment:
method says `updateDatabaseFromHeaders` but you're actually updating fields, what do you think of moving these 2 somewhere else?